src/test/directives/run-async_test.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {InitialStateError, runAsync} from '../../directives/run-async.js';
+import {render} from '../../lib/render.js';
+import {html} from '../../lit-html.js';
+import {stripExpressionMarkers} from '../test-utils/strip-markers.js';
+
+const assert = chai.assert;
+
+const hasAbortController = typeof AbortController === 'function';
+const testIfHasAbortController = hasAbortController ? test : test.skip;
+
+suite('runAsync', () => {
+  let container: HTMLDivElement;
+
+  setup(() => {
+    container = document.createElement('div');
+  });
+
+  const renderTest =
+      (key: string,
+       f: (s: string, options: {signal?: AbortSignal}) => Promise<string>) =>
+          render(
+              html`${runAsync(key, f, {
+                success: (s) => html`Success: ${s}`,
+                pending: () => 'Pending',
+                initial: () => `Initial`,
+                failure: (e: Error) => `Error: ${e.message}`
+              })}`,
+              container);
+
+  test('renders pending then success templates', async () => {
+    renderTest('test', async (s: string) => s);
+    assert.equal(stripExpressionMarkers(container.innerHTML), 'Pending');
+    await 0;
+    assert.equal(stripExpressionMarkers(container.innerHTML), 'Success: test');
+  });
+
+  test('renders pending then initial templates', async () => {
+    renderTest('test', async (_: string) => {
+      throw new InitialStateError();
+    });
+    assert.equal(stripExpressionMarkers(container.innerHTML), 'Pending');
+    await 0;
+    assert.equal(stripExpressionMarkers(container.innerHTML), 'Initial');
```

Comment:
Make `runAsync` wait a microtask after running the async fn to get a shot at rendering initial before pending?